hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,258 @@
+				failedTest{
+					flakeRate: flakerate,
+					env:       env,
+					testName:  test,
+				})
+		}
+
+		sort.Slice(failedTestList, func(i, j int) bool {
+			return failedTestList[i].flakeRate < failedTestList[j].flakeRate
+		})
+		envFailedTestList[env] = failedTestList
+	}
+	// we convert the result into a 2d string slice representing a markdown table,
+	// whose each line represents a line of the table
+	table := [][]string{
+		// title of the table
+		{"Environment", "Test Name", "Flake Rate"},
+	}
+	// if an env has too much failures we will just skip it and print a message in the end
+	// if the failed tests have high flake rates(over 50% for all), it will also be skipped in the table
+	foldedFailures := []string{}
+	for env, list := range envFailedTestList {
+		if len(list) > maxItemEnv {
+			foldedFailures = append(foldedFailures, env)
+			continue
+		}
+		for i, item := range list {
+			if item.flakeRate > 50 {
+				if i == 0 {
+					// if this is the first failed test
+					// that means each tests in this env
+					// has a flakerate>50% and all of them will
+					// not be shown in the table
+					foldedFailures = append(foldedFailures, env)
+				}
+				break
+			}
+			flakeRateString := fmt.Sprintf("%.2f%% %s", item.flakeRate, testFlakeChartMDLink(env, item.testName))
+			if item.flakeRate == unknown {
+				flakeRateString = "Unknown"
+			}
+			table = append(table, []string{
+				envChartMDLink(env, len(list)),
+				item.testName + gopoghMDLink(pr, rootJob, env, item.testName),
+				flakeRateString,
+			})
+		}
+	}
+
+	builder := strings.Builder{}
+	builder.WriteString(
+		fmt.Sprintf("Here are the number of top %d failed tests in each environments with lowest flake rate.\n\n", maxItemEnv))
+	builder.WriteString(generateMarkdownTable(table))
+	if len(foldedFailures) > 0 {
+
+		builder.WriteString("\n\n Besides the following environments also have failed tests:")
+		for _, env := range foldedFailures {
+			builder.WriteString(fmt.Sprintf("\n\n - %s: %d failed %s ", env, len(envFailedTestList[env]), gopoghMDLink(pr, rootJob, env, "")))
+		}
+	}
+	builder.WriteString("\n\nTo see the flake rates of all tests by environment, click [here](https://minikube.sigs.k8s.io/docs/contrib/test_flakes/).")
+	return builder.String()
+}
+func envChartMDLink(env string, failedTestNumber int) string {
+	return fmt.Sprintf("[%s (%d failed)](https://gopogh-server-tts3vkcpgq-uc.a.run.app/?env=%s)", env, failedTestNumber, env)
+}
+
+func testFlakeChartMDLink(env string, testName string) string {
+	return fmt.Sprintf("[(chart)](https://gopogh-server-tts3vkcpgq-uc.a.run.app/?env=%s&test=%s)", env, testName)
+}
+
+func gopoghMDLink(pr, rootJob, env, testName string) string {
+	return fmt.Sprintf("[(gopogh)](https://storage.googleapis.com/minikube-builds/logs/%s/%s/%s.html#%s)", pr, rootJob, env, testName)
+}
+
+// generateMarkdownTable convert 2d string slice into markdown table. The first string slice is the header of the table
```

Comment:
```suggestion
// generateMarkdownTable convert 2d string slice into markdown table. The first
// string slice is the header of the table
```
Shortening long comments for better readability